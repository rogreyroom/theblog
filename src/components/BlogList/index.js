import React, { useEffect } from 'react'

import { useSelector, useDispatch } from 'react-redux'

import { fetchCommentsData } from '../../redux/comments/commentsActions'
import { fetchUsersData } from '../../redux/users/usersActions'
import PostCard from '../PostCard'
import classes from './index.module.scss'

const BlogList = () => {
  const { posts } = useSelector((state) => state.posts)
  const dispatch = useDispatch()

  useEffect(() => {
    if (posts.length > 0) {
      dispatch(fetchCommentsData())
      dispatch(fetchUsersData())
    }
  }, [dispatch, posts])

  if (posts.length === 0) return <h2>Loading...</h2>

  return (
    <section className={classes.posts}>
      <h2 className={classes.posts__title}>
        Blog posts <span>/ {posts.length} /</span>
      </h2>
      {posts.length > 0 && (
        <ul className={classes.posts__list}>
          {posts.map((post) => (
            <li className={classes.posts__item} key={post.id}>
              <PostCard id={post.id} title={post.title} key={post.id} />
            </li>
          ))}
        </ul>
      )}
    </section>
  )
}

export default BlogList
