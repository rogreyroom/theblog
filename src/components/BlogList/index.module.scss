.posts {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  &__title {
    color: var(--c-black);
    font-size: clamp(var(--fs-header-small), 0.4483rem + 2.7586vw,  var(--fs-header-large));
    margin: var(--normal) 0;

    & span {
      color: var(--c-grey);
      font-size: clamp(var(--fs-text-small), 0.8060rem + 0.3448vw,  var(--fs-text-normal));
    }
  }
  &__list {
    list-style: none;
    margin: 0;
    padding: 0;
    width: 100%;
  }
  &__item {
    display: flex;
    justify-content: center;
  }
}