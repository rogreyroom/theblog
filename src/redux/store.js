import { configureStore } from '@reduxjs/toolkit'
import postsSlice from './posts/postsSlice'
import commentsSlice from './comments/commentsSlice'

const store = configureStore({
  reducer: {
    posts: postsSlice.reducer,
    comments: commentsSlice.reducer,
  },
})

export default store
