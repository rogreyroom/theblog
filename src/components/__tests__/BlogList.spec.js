/**
 * @jest-environment jsdom
 */
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { screen, render, cleanup } from '@testing-library/react'
import { Provider } from 'react-redux'
import BlogList from '../BlogList'
import store from '../../redux/store'

describe('BlogList component', () => {
  let element
  beforeEach(() => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <BlogList />
        </Router>
      </Provider>,
    )
    element = container
  })

  it('renders the component', () => {
    expect(element).toMatchSnapshot()
  })

  it('component contains text "Loading..." when no data is loaded', () => {
    expect(screen.getByText(/loading/i)).toHaveTextContent('Loading...')
  })

  afterAll(cleanup)
})
