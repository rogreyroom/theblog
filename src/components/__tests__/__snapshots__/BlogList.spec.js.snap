// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogList component renders the component 1`] = `
<div>
  <h2>
    Loading...
  </h2>
</div>
`;
